state-synchronizer/src/tests/fuzzing.rs

```diff
@@ -47,18 +47,28 @@ proptest! {
 }
 
 pub fn test_state_sync_msg_fuzzer_impl(msg: StateSynchronizerMsg) {
-    // start up coordinator
-    let (_coordinator_sender, coordinator_receiver) = mpsc::unbounded();
-    let (mempool_sender, _mempool_receiver) = mpsc::channel(1_024);
-    let config = NodeConfig::default_for_validator();
+    // Generate a genesis change set
+    let (genesis, _) = vm_genesis::test_genesis_change_set_and_validators(Some(1));
+
+    // Create test diem database
+    let db_path = diem_temppath::TempPath::new();
+    db_path.create_as_dir().unwrap();
+    let (db, db_rw) = DbReaderWriter::wrap(DiemDB::new_for_test(db_path.path()));
```

Comment:
so from quick glance, this uses a real rocksdb db right : o ?

If so two thoughts:

1. that's great! I think we can really increase coverage like that
2. but your speed issue probably come from that. I'd recommend either doing another fuzzer where you use the real executor proxy with a mock db, that should be much faster and I think we already exercise some of these DIemDB path in other fuzzers, or create two fuzzers: one that does that, and on that does what I just say. One will be slow with good coverage, the other will be faster will less coverage. 

wdyt?

Reply:
Thanks @mimoo!

Yeah, this fuzzer uses a real database for testing (read-write mode, without the pruner): https://github.com/diem/diem/blob/44e0fd708869d12511eb44d898e6838d116899c8/storage/diemdb/src/lib.rs#L217

> I'd recommend either doing another fuzzer where you use the real executor proxy with a mock db, that should be much faster and I think we already exercise some of these DIemDB path in other fuzzers, or create two fuzzers: one that does that, and on that does what I just say. One will be slow with good coverage, the other will be faster will less coverage.

Yeah, I think this would make sense and it's something we can do in the future. The only issue with having a real executor proxy and a mock database right now is that the MockStorage as currently provided by the state sync code doesn't actually implement the DB interface: it's designed to be used by the MockExecutorProxy, so we can't just swap out the DB 😕 

My preference would be to land this for now (so we get *some* real coverage), and then after I've finished refactoring all the tests (and hopefully end up removing a lot of these mocked objects), I can implement a Mock DB that implements the real DB interface, so we can swap them out and get the best of both. Does that make sense?

Reply:
> doesn't actually implement the DB interface

which is what we were doing before anyway no? Even if we return errors, or empty responses, we still exercise that path : o

but yeah no problem approving this and re-visiting later, let's get that in!

Reply:
Yeah, let's sync with this once I've got the tests in a better position 😄 I'm sure we can do something clean.